### PR TITLE
[api] API v2.2.0 Documentation

### PR DESCRIPTION
**Pull request by abellotti**
- Updated Versioning page to talk about v2.2.0 and its features
- Updated Collection page to include the new collections
- Enhanced Filtering:
 * * wildcard
 * nil or NULL equality/inequality match
 * Virtual attribute including one-level association
- Case insensitive sort option
- Group Management
- User Management
- Host Management - Queries and Password updates.
- Service Reconfiguration
- Additional actions on VMs
- Instance Management
- Approving & Denying Provision and Automation Request
- Supporting DELETE of own authentication token
- Adding support for Tenant Quota CRUD
- Accessing vms subcollection of services
- Adding support for Service Orders (aka _Shopping Carts_)
- Refresh Dialog fields

### CODE REVIEW

**Comment by abellotti** [comment]
@mfeifer @imtayadeway for your reviewing pleasures.
cc/ @Fryguy

File: api/reference/groups.adoc [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+* link:#querying-groups[Querying Groups]
+* link:#creating-groups[Creating Groups]
+* link:#editing-groups[Editing Groups]
+* link:#deleting-groups[Deleting Groups]
+
+[[querying-groups]]
+=== Querying Groups
+
+Querying all groups in the system is simply:
+
```
Comment (mfeifer) [inline api/reference/groups.adoc]: remove "is simply"

File: api/reference/instances.adoc [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+[source,json]
+----
+{
+  "action" : "start"
+}
+----
+
+
+Requests can also be made on multiple instances by targetting the primary collection:
```
Comment (mfeifer) [inline api/reference/instances.adoc]: targeting

File: api/reference/instances.adoc [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+[source,json]
+----
+{
+  "action" : "pause"
+}
+----
+
+[[terminate-instance]]
+=== Terminating and Instance
```
Comment (mfeifer) [inline api/reference/instances.adoc]: "an" not "and"

File: api/reference/instances.adoc [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+[source,json]
+----
+{
+  "action" : "reboot-guest"
+}
+----
+
+[[reset-instance]]
+=== Reset an Instance
```
Comment (mfeifer) [inline api/reference/instances.adoc]: Resetting

File: api/reference/service_orders.adoc [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+      "Basic_Host_Name" : "super_vm_b",
+      "Basic_Host_Ip" : "192.168.120.102"
+    },
+    ...
+  ]
+}
+----
+
+
+Create a wish list service_order:
```
Comment (mfeifer) [inline api/reference/service_orders.adoc]: Creating

**Comment by mfeifer** [comment]
Minor nit - decide if you want to use the gerund of a verb such as "Resetting" or the verb itself such as "Reset" for titles.  If you don't want to go through and change them all, now, that's fine.

**Comment by abellotti** [comment]
Thanks @mfeifer, made those changes and repushed, also the titles were not consistent, went with the gerund variant.

File: api/appendices/resource_attributes.adoc [hunk 6]
```diff
@@ -74,6 +75,33 @@ POST /api/rates
...
+POST /api/groups
+----
+
+==== Required attributes:
+
+[cols="2,1,3",options="header",width="100%"]
+|=====================
+| Attribute | Type | Description
+| description | string | Description
+| role | hash | Role the group is assigned to, requires an *href*, *id* or *name* attribute in the hash
```
Comment (imtayadeway) [inline api/appendices/resource_attributes.adoc]: Have we used "hash" everywhere? I wonder if it's better to use "object" to describe these, as it's JSON?

File: api/overview/auth.adoc [hunk 7]
```diff
@@ -120,6 +120,31 @@ Content-Type: application/json
...
+       -H "X-Auth-Token: 93d98fd44cc5a401ff385ba87b21fd13"
+       http://localhost:3000/api/auth
+----
+
+[[response]]
+Response for the token DELETE request:
+
+[source,data]
+----
+HTTP/1.1 204 OK
```
Comment (imtayadeway) [inline api/overview/auth.adoc]: Is this not "No Content"?

File: api/reference/services.adoc [hunk 8]
```diff
@@ -304,6 +360,30 @@ POST /api/services/:id
 }
 ----
 
+[[service-reconfiguring]]
+==== Service Reconfiguring
+
+When a service is defined with an Reconfigure Automation Entrypoint, the _reconfigure_ action is shown
```
Comment (imtayadeway) [inline api/reference/services.adoc]: s/an/a/

File: api/appendices/resource_attributes.adoc (hunk 6 above)
Comment (abellotti) [inline api/appendices/resource_attributes.adoc]: Thanks Tim, I've updated and pushed up the others. For this (and all *hash* types here), Object is not necessarily always true, as our own hashes are for referencing only. We do mention the word *struct* for these in the Overview/Json Convention file.  We can update all to struct but I'm not sure   it's be clearer.

**Comment by mfeifer** [comment]
@imtayadeway are you okay with the changes on a technical level?  If so, I can merge.

**Comment by imtayadeway** [comment]
@mfeifer GTG :thumbsup:

**Pull request merged**